packages/experimental/kai/src/app/ShellProvider.tsx

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import React, { FC, PropsWithChildren, useCallback, useEffect, useMemo, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { ShellDisplay, ShellLayout } from '@dxos/client';
+import { MemoryShellRuntime } from '@dxos/client-services';
+import { useConfig, useCurrentSpace, useIdentity } from '@dxos/react-client';
+import { mx } from '@dxos/react-components';
+import { Shell } from '@dxos/react-ui';
+
+import { ShellContext } from '../hooks';
+
+/**
+ * Renders the DXOS shell and provides a way to set the layout of the shell from the rest of the app.
+ */
+// TODO(wittjosiah): Factor out?
```

Comment:
Yeah, feel like app-kit? or some other "patterns"?

Reply:
Probably into react-ui alongside the Shell component itself I think

Reply:
💯 Agreed